symbolic-symcache/src/writer.rs

```diff
@@ -294,16 +295,17 @@ where
         // requires some address range. Since we can't possibly know the actual size, just assume
         // that the symbol is VERY large.
         let size = match symbol.size {
-            0 => !0,
-            s => s,
+            0 => u16::MAX,
+            s => s as u16,
```

Comment:
Only noticing this now: This will not clamp to `u16::MAX` and rather wrap around. Can you add the `cmp::min` back?

Reply:
Yes, I made a mistake while testing this. I'll add it back.